source/slang/slang-check-expr.cpp

```diff
@@ -1355,6 +1355,162 @@ namespace Slang
         }
     }
 
+    RefPtr<Expr> SemanticsVisitor::CheckMatrixSwizzleExpr(
+        MemberExpr* memberRefExpr,
+        RefPtr<Type>      baseElementType,
+        IntegerLiteralValue baseElementRowCount,
+        IntegerLiteralValue baseElementColCount)
+    {
+        RefPtr<MatrixSwizzleExpr> swizExpr = new MatrixSwizzleExpr();
+        swizExpr->loc = memberRefExpr->loc;
+        swizExpr->base = memberRefExpr->baseExpression;
+
+        int elementIndices[4][2];
+        int elementCount = 0;
+
+        bool anyDuplicates = false;
+        int oneIndexed = -1;
+
+        String swizzleText = getText(memberRefExpr->name);
+        auto coursor = swizzleText.begin();
+
+        while (coursor != swizzleText.end())
+        {
+            // Throw out swizzling with more than 4 output elements
+            if (elementCount >= 4) {
```

Comment:
Coding conventions can be somewhat arbitrary - and the { on the same line has some good properties, like making better use of vertical space. On the Slang project we've gone with the { } on the same indentation.

Reply:
Within our coding standards the opening brace indentation should line up with the closing brace, so we need to have that fixed.

Reply:
In case the phrasing isn't clear: the Slang codebase prefers the "Allman" convention:

```c++
if(condition)
{
    stuff();
}
```

and does not use the "K&R" convention:

```c++
if(condition) {
    stuff();
}
```

The main exception is when you can reduce a block to a single line:

```c++
if(condition) { stuff(); }
```